api-service/database/licenses.go

```diff
@@ -17,100 +17,114 @@ package database
 
 import (
 	"context"
+	"errors"
 	"time"
 
 	"github.com/amreo/mu"
 	"github.com/ercole-io/ercole/utils"
 	"go.mongodb.org/mongo-driver/bson"
 )
 
-// ListLicenses list licenses
-func (md *MongoDatabase) ListLicenses(full bool, sortBy string, sortDesc bool, page int, pageSize int, location string, environment string, olderThan time.Time) ([]interface{}, utils.AdvancedErrorInterface) {
-	var out []interface{} = make([]interface{}, 0)
-	//Find the informations
+// SearchLicenses search licenses
+func (md *MongoDatabase) SearchLicenses(mode string, sortBy string, sortDesc bool, page int, pageSize int, location string, environment string, olderThan time.Time) ([]interface{}, utils.AdvancedErrorInterface) {
+	var isFull bool
+	if mode == "full" {
+		isFull = true
+	} else if mode == "summary" {
+		isFull = false
+	} else {
+		return nil, utils.NewAdvancedErrorPtr(errors.New("Wrong mode value"), "")
```

Comment:
Why don't you have specified the error class???

Reply:
I didn't know which kind of error class use! I forgot to ask!
Which one is more appropriate?

Reply:
:+1: